Deploying_Seq2Seq_Model_with_Hybrid_Frontend.ipynb

```diff
@@ -0,0 +1,720 @@
+    "\n",
+    "\n",
+    "### Decoder\n",
+    "\n",
+    "We perform the same process for tracing the decoder as we did for the encoder. Notice that we call forward on a set of random inputs to the traced_encoder to get the output that we need for the decoder. This is not required, as we could also simply create a tensor of the correct shape, type, and value range. We trace the model in the exact fashion that we traced the encoder.\n",
+    "\n",
+    "\n",
+    "### GreedySearchDecoder\n",
+    "\n",
+    "Recall that we scripted our searcher module due to the presence of data-dependent control flow. In the case of scripting, we do the conversion work up front by adding the decorator and making sure the implementation complies with scripting rules. We initialize the scripted searcher the same way that we would initialize an un-scripted variant."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "### Convert encoder model\n",
+    "# Create artificial inputs\n",
+    "test_seq = torch.LongTensor(MAX_LENGTH, 1).random_(0, voc.num_words)\n",
+    "test_seq_length = torch.LongTensor([test_seq.size()[0]])\n",
+    "# Trace the model\n",
+    "traced_encoder = torch.jit.trace(encoder, (test_seq, test_seq_length))\n",
+    "\n",
+    "### Convert decoder model\n",
+    "# Create and generate artificial inputs\n",
+    "test_encoder_outputs, test_encoder_hidden = traced_encoder(test_seq, test_seq_length)\n",
+    "test_decoder_hidden = test_encoder_hidden[:decoder.n_layers]\n",
+    "test_decoder_input = torch.LongTensor(1, 1).random_(0, voc.num_words)\n",
+    "# Trace the model\n",
+    "traced_decoder = torch.jit.trace(decoder, (test_decoder_input, test_decoder_hidden, test_encoder_outputs))\n",
+    "\n",
+    "### Initialize searcher module\n",
+    "scripted_searcher = GreedySearchDecoder(traced_encoder, traced_decoder, decoder.n_layers)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Print Graphs\n",
+    "\n",
+    "Now that our models are in Torch Script form, we can print the graphs of each to ensure that we captured the computational graph appropriately."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('encoder graph', traced_encoder.graph)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('decoder graph', traced_decoder.graph)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('searcher graph', scripted_searcher.graph)"
```

Comment:
You could get away with only showing the graph for the `scripted_searcher` instance, because the graphs for the encoder and decoder will be directly inlined.